.github/workflows/self-tests-ubuntu-22-04.yaml

```diff
@@ -0,0 +1,91 @@
+name: Ubuntu 22.04 self-tests
```

Comment:
This is _not_ the way to run self-tests on multiple versions of Ubuntu. Creating a new workflow file creates a large amount of code duplication. We also don't want things to diverge between the different Ubuntu-target self-test workflows, and thus have to manage twice as many workflows. Look at using a [matrix](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#using-a-matrix-strategy).